### DISCUSSION TITLE
How does the BirdNET-Analyzer --overlap option work?

### DISCUSSION THREAD

**Discussion by ceperman** [birdnet-team/BirdNET-Analyzer#434] [General]
I'm not sure exactly how the --overlap option works. The guide says:
`--overlap, Overlap of prediction segments. Values in [0.0, 2.9]. Defaults to 0.0.`

By default, analysis proceeds in 3 sec non-overlapping sections, so:
0.0-3.0, 3.0-6.0, 6.0-9.0 etc
If I set overlap to 1.0, does this mean additional 3 sec sections at 1.0 sec intervals, thus:
0.0-3.0, 1.0-4.0, 2.0-5.0, 3.0-6.0, 4.0-7.0 etc ?
If so, what happens if the overlap is not a dividend of 3.0 secs?

**Comment by kahst** [comment]
An overlap of 1.0 seconds means that two consecutive segments overlap by 1 second. So the resulting timestamps for predictions would be:

`0.0-3.0, 2.0-5.0, 4.0-7.0, 6.0-9.0`and so on. The maximum overlap is 2.9 seconds, which would result in a "hop length" of 0.1 seconds.

In other words, a 1.0s overlap results in a 2.0s hop length/step size.

**Reply by ceperman** [reply]
Many thanks. Perhaps a bit more explanation in the usage guide?
